src/sentry/models/debugfile.py

```diff
@@ -487,28 +524,18 @@ def detect_dif_from_path(path):
     else:
         objs = []
         for obj in archive.iter_objects():
-            data = {
-                'type': obj.kind,
-                'features': list(obj.features),
-            }
-            objs.append((obj.file_format, obj.arch, obj.debug_id, path, data))
+            objs.append(DifMeta.from_object(obj, path, name=name))
         return objs
 
 
-def create_debug_file_from_dif(to_create, project, overwrite_filename=None):
+def create_debug_file_from_dif(to_create, project):
```

Comment:
`overwrite_filename` seems to have been dead code. It's still possible however by passing `name=...` to `detect_dif_from_path`. Eventually, `symbolic` will attempt to read the actual original name from the debug file where it's possible (e.g. from PDBs and some dSYMs).